src/ray/ray_impl.rs

```diff
@@ -175,6 +177,36 @@ impl<T: Scalar + Copy, const D: usize> Ray<T, D> {
     }
 }
 
+
+impl<const D: usize> Ray<f32, D> {
+/// Intersect [`Aabb`] by [`Ray`]
+    /// Returns slice of intersections, two numbers [`f32`]
+    /// where first number is distance from [`Ray`] to nearest intersection point,
+    /// second number is distance from [`Ray`] to fathest intersection point
+    /// 
+    /// If no instersections returns infinity for both distancies
+    pub fn intersection_slice_for_aabb(&self, aabb: &Aabb<f32, D>) -> (f32, f32)
+    {
+        // https://iquilezles.org/articles/intersectors/
+        let mut n = self.origin.coords - aabb.center().coords;
+        n.component_mul_assign(&self.inv_direction);
+
+        let k = self.inv_direction.abs().component_mul(&(aabb.size() * 0.5));
```

Comment:
Where does this box `size() * 0.5` come from? The original algorithm linked does not have it. Is it because `boxSize` referred to there are the half extents?

Reply:
in linked algo author represent box as center aligned to zero.
![image](https://user-images.githubusercontent.com/64865196/236536559-5233bd82-a696-4259-87f6-c2c2831af2a3.png)

So, boxSize in this algo is not width and height of box but distances to edges / absolute coords of planes of box

Reply:
Yeah okay that makes sense